addons/web_editor/static/src/js/editor/odoo-editor/src/utils/sanitize.js

```diff
@@ -78,6 +79,38 @@ export function areSimilarElements(node, node2) {
     );
 }
 
+/**
+ * Returns a URL if link's label is a valid email of http URL, null otherwise.
+ *
+ * @param {HTMLAnchorElement} link
+ * @returns {String|null}
+ */
+function deduceURLfromLabel(link) {
+    const label = link.innerText.trim();
+    // Check first for e-mail.
+    let match = label.match(EMAIL_REGEX);
+    if (match) {
+        return match[1] ? match[0] : 'mailto:' + match[0];
+    }
+    // Check for http link.
+    // Regex with 'g' flag is statefull, reset lastIndex before and after use.
+    URL_REGEX_WITH_INFOS.lastIndex = 0;
+    match = URL_REGEX_WITH_INFOS.exec(label);
+    URL_REGEX_WITH_INFOS.lastIndex = 0;
```

Comment:
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/lastIndex#avoiding_side_effects

Reply:
When I saw that I thought "omg but we need to do that everywhere else we use this regex", but then I realized this is only when called with `exec`. I think the comment should mention that. Also, "stateful" only takes one "l".

Reply:
Changed the comment a bit.